pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -114,6 +115,8 @@ func limitsChanged(existing []corev1.LimitRangeItem, toUpdate []corev1.LimitRang
 }
 
 func (c *SyncController) CreateResourceQuota(ns *corev1.Namespace) (runtime.Object, error) {
+	c.ensureContainerDefaultLimitsAnnotationSet(ns)
```

Comment:
Calling this function here seems a little out of place to me, because this function is called `CreateResourceQuota` but this is about limit ranges. There's another method `createLimitRange` called just after in the sync handler that might be more appropriate. That way you could also deduplicate some effort by using the result of the `getResourceLimitToUpdate` call to get the default and then calling something similar to `validateAndSetNamespaceQuota` to set it.

Reply:
Agreed! Can't we just call the function first thing in the `createLimitRange` method? The new function just sets the annotation, and all the other logic takes care of the actual LimitRange creation. The only thing duplicated is the call to `getProjectContainerDefaultLimit`, which might be OK? Sorry, maybe I am missing something - not seeing how to refactor it more while avoiding restructuring more things.